### DISCUSSION TITLE
gatsby - OpenWeatherMap

### DISCUSSION THREAD

**Discussion by flynnend** [community/community#59593] [Other Feature Feedback, Questions, & Ideas] [answered]
Question

Hi everyone,

I’m having trouble with OpenWeatherMap in Gatsby. Could someone help me out? I want to use the gatsby-source-openweathermap source plugin, but I’m not sure how to configure it. Could someone show me an example configuration? Thanks in advance for your help!

**Comment by TEHAQUE** [comment, marked as answer]
Hello,

I encountered a similar issue in my own project. I found that incorporating the react-leaflet library resolved the problem for me. Additionally, make sure to include the plugin configuration in your gatsby-config.js file. To gain better visibility into any errors and facilitate potential hosting on Gatsby Cloud, I recommend setting up SSR (Server Side Rendering) in the gatsby-config.js file. This will allow you to quickly identify and address any specific issues that may arise during the integration process.

This is my gatsby-config.js file:
```javascript
* @type {import('gatsby').GatsbyConfig}
 */
module.exports = {
  siteMetadata: {
    title: `your-app`,
    siteUrl: `https://www.yourdomain.tld`
  },
  flags: {
    DEV_SSR: true,
  },
  plugins: [
    "gatsby-plugin-sass",
    {
        resolve: 'gatsby-plugin-react-leaflet',
        options: {
          linkStyles: true 
        }
    }
  ]
};
``` 
Make sure to install `gatsby-plugin-react-leaflet` plugin.

I hope this information proves helpful in resolving the problems you're facing with the OpenWeatherMap plugin in Gatsby. If you have any further questions, feel free to ask.

**Reply by flynnend** [reply]
That's it, thanks for your help. I can contact you more privately because I have a few questions?

**Reply by TEHAQUE** [reply]
Sure, I'll add you to one of the ropos. We can chat in the discussions tab.
